src/OpenSSL/SSL.py

```diff
@@ -1688,6 +1694,17 @@ def set_tlsext_host_name(self, name):
         # XXX I guess this can fail sometimes?
         _lib.SSL_set_tlsext_host_name(self._ssl, name)
 
+    @_requires_x509_verify
```

Comment:
What do you think about integrating this into `set_tlsext_host_name`, so there's only a single method call required for correct/secure usage?

Reply:
Certainly doable, but the main difference with `set_tlsext_host_name` is that it should not be called on IP addresses, per RFC6066, Section 3

Reply:
Excellent point, you have to use ``X509_VERIFY_PARAM_set1_ip()`` for IP address validation any way. Feel free to copy my implementation from Python's ssl module. The function ``_ssl_configure_hostname()`` takes care of SNI, hostname verification, and special cases IP addresses.

https://github.com/python/cpython/blob/ba251c2ae6654bfc8abd9d886b219698ad34ac3c/Modules/_ssl.c#L862-L926